src/client/components/pages/collection.js

```diff
@@ -224,6 +232,59 @@ class CollectionPage extends React.Component {
 		}, this.pagerElementRef.triggerSearch);
 	}
 
+	handleSubscribe() {
```

Comment:
As far as I can tell, all this code (`handleSubscribe`, `handleUnsubscribe`, `setIsSubscribed` and the buttons that go with it) is duplicated for each display page. Instead it would be better if the whole subscription part was a separate component that can then be added to the relevant pages.
You'll probably want to pass the submissionUrl as a prop to make it more flexible.

We might end up with separate CollectionSubscription  and an EntitySubscription components if they require very different implementations.